tests/src/3d/testqgs3drendering.cpp

```diff
@@ -1408,5 +1471,274 @@ void TestQgs3DRendering::testFilteredExtrudedPolygons()
+  grayImage = convertDepthImageToGray16Image( depthImage );
+  QVERIFY( renderCheck( "depth_wheel_action_4", grayImage, 550 ) );
+
+  scene->cameraController()->depthBufferCaptured( depthImage );
+
+  QGSCOMPARENEARVECTOR3D( testCam->zoomPoint(), QVector3D( -32.3, 221.7, 183.2 ), 1.0 );
+  QGSCOMPARENEARVECTOR3D( testCam->cameraPose()->centerPoint(), QVector3D( -32.3, 221.7, 183.2 ), 1.0 );
+  QGSCOMPARENEAR( testCam->cameraPose()->distanceFromCenterPoint(), 101.1, 0.1 );
+  QCOMPARE( testCam->cumulatedWheelY(), 0 );
+
+  // Checking camera position
+  QVector3D diff = scene->cameraController()->camera()->position() - startPos;
+  QGSCOMPARENEARVECTOR3D( diff, QVector3D( 125, -850, -700 ), 3.0 );
+
+  delete scene;
+  mapSettings.setLayers( {} );
+  demTerrain->deleteLater();
+}
+
+void TestQgs3DRendering::testDebugMap()
+{
+  // =============================================
+  // =========== creating Qgs3DMapSettings
+  const QString dataDir( TEST_DATA_DIR );
+  QgsRasterLayer *layerDtm = new QgsRasterLayer( dataDir + "/3d/dtm.tif", "dtm", "gdal" );
+  QVERIFY( layerDtm->isValid() );
+
+  const QgsRectangle fullExtent = layerDtm->extent();
+
+  QgsProject project;
+  project.addMapLayer( layerDtm );
+
+  Qgs3DMapSettings mapSettings;
+  mapSettings.setCrs( project.crs() );
+  mapSettings.setExtent( fullExtent );
+  mapSettings.setLayers( {layerDtm, mLayerBuildings} );
+
+  mapSettings.setTransformContext( project.transformContext() );
+  mapSettings.setPathResolver( project.pathResolver() );
+  mapSettings.setMapThemeCollection( project.mapThemeCollection() );
+
+  QgsDemTerrainGenerator *demTerrain = new QgsDemTerrainGenerator();
+  demTerrain->setLayer( layerDtm );
+  mapSettings.setTerrainGenerator( demTerrain );
+  mapSettings.setTerrainVerticalScale( 3 );
+
+  QgsPointLightSettings defaultPointLight;
+  defaultPointLight.setPosition( QgsVector3D( 0, 400, 0 ) );
+  defaultPointLight.setConstantAttenuation( 0 );
+  mapSettings.setLightSources( {defaultPointLight.clone() } );
+  mapSettings.setOutputDpi( 92 );
+
+  // =========== creating Qgs3DMapScene
+  QPoint winSize = QPoint( 640, 480 ); // default window size
+
+  QgsOffscreen3DEngine engine;
+  engine.setSize( QSize( winSize.x(), winSize.y() ) );
+  Qgs3DMapScene *scene = new Qgs3DMapScene( mapSettings, &engine );
+  engine.setRootEntity( scene );
+
+  // =========== set camera position
+  scene->cameraController()->setLookingAtPoint( QVector3D( 0, 0, 0 ), 1500, 40.0, -10.0 );
+
+  // =========== activate debug depth map
+  mapSettings.setDebugDepthMapSettings( true, Qt::Corner::BottomRightCorner, 0.5 );
+
+  QImage img = Qgs3DUtils::captureSceneImage( engine, scene );
+  QVERIFY( renderCheck( "debug_map_1", img, 100, 15 ) );
+
+  // =========== activate debug shadow map
+  mapSettings.setDebugShadowMapSettings( true, Qt::Corner::TopLeftCorner, 0.5 );
+
+  img = Qgs3DUtils::captureSceneImage( engine, scene );
+  QVERIFY( renderCheck( "debug_map_2", img, 100, 15 ) );
```

Comment:
This doesn't seem to actually test anything: `expected_debug_map_1.png` and `expected_debug_map_2.png` are binary equal.

If you make proper rendering checks for the _debugShadowMap_ and _debugDepthMap_ windows you could also eliminate the need for `convertDepthImageToGray16Image()` and replace it with one check against a map with `mapSettings.setDebugDepthMapSettings( true, Qt::Corner::BottomRightCorner, **1** );`

Reply:
In fact the test fails because the 2 images are the same. It was my mistake to copy the expected image from another test to have a default image to check against but in fact the `captureSceneImage` failed to render the debug maps

Reply:
Hi @uclaros, I have been looking at this issue all day but I can not find why the debug texture map is not working at all. Any clues?

Reply:
Hmm, I don't know, I would need to investigate. It could be related to it being on a different viewport, so I would start by checking that the 3dAxis (which is also on a different viewport) is being exported.